### ISSUE TITLE
Interrupted system call

### ISSUE THREAD

**Issue by Chrysweel** [progval/Limnoria#1093]
I had a disconnection of my bot, and I dont know the reason.
Somebody know it ?

WARNING 2015-04-14T16:05:06 supybot Disconnect from irc.my_network.net:6667: error: [Errno 4] Interrupted system call.

```
ERROR 2015-04-14T16:05:06 supybot Uncaught exception in in drivers.run:
Traceback (most recent call last):
  File "/home/user/.local/lib/python2.7/site-packages/supybot/drivers/__init__.py", line 105, in run
    driver.run()
  File "/home/user/.local/lib/python2.7/site-packages/supybot/drivers/Socket.py", line 191, in run
    time.sleep(conf.supybot.drivers.poll())
  File "/home/user/.local/lib/python2.7/site-packages/supybot/registry.py", line 411, in __call__
    self.set(_cache[self._name])
  File "/home/user/.local/lib/python2.7/site-packages/supybot/registry.py", line 460, in set
    self.setValue(float(s))
  File "/home/user/.local/lib/python2.7/site-packages/supybot/registry.py", line 478, in setValue
    super(PositiveFloat, self).setValue(v)
  File "/home/user/.local/lib/python2.7/site-packages/supybot/registry.py", line 466, in setValue
    super(Float, self).setValue(float(v))
  File "/home/user/.local/lib/python2.7/site-packages/supybot/registry.py", line 375, in setValue
    callback(*args, **kwargs)
  File "/home/user/.local/lib/python2.7/site-packages/supybot/drivers/Socket.py", line 321, in setTimeout
    self.conn.settimeout(conf.supybot.drivers.poll())
  File "/usr/lib/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
  File "/usr/lib/python2.7/socket.py", line 170, in _dummy
    raise error(EBADF, 'Bad file descriptor')
error: [Errno 9] Bad file descriptor
ERROR 2015-04-14T16:05:06 supybot Exception id: 0x8a657
```

And after I have in log this line again and again

> ERROR 2015-04-14T16:05:06 supybot Schedule is the only remaining driver, why do we continue to live?

**Comment by ProgVal** [comment]
Looks like it will be fixed in Python 3.5: https://www.python.org/dev/peps/pep-0475/

**Comment by progval** [comment]
Limnoria stopped supporting these Python versions years ago, so let's consider this fixed
